library/Datastore.nucleus.js

```diff
@@ -913,6 +917,11 @@ class NucleusDatastore {
               } else {
                 // If the property is a number, collapse as an array.
                 if (nucleusValidator.isArray(accumulator)) {
+                  //if array is not emty
```

Comment:
I think you can replace this to `if (nucleusValidator.isArray(accumulator) && value) {`